tracing-core/src/callsite.rs

```diff
@@ -169,3 +177,169 @@ impl Hash for Identifier {
         (self.0 as *const dyn Callsite).hash(state)
     }
 }
+
+/// An intrusive atomic push only linked-list.
+struct LinkedList {
+    head: AtomicPtr<Registration>,
+}
```

Comment:
i'm not _totally_ convinced we need a separate `LinkedList` struct with methods on it, since it's only ever used as part of the callsite registry. for example, we have a `for_each` method on `LinkedList` that takes an arbitrary closure, but we only ever call it in one place and only ever pass a single closure to it...so it _could_ just be part of the `rebuild_interest` function...

on the other hand, it is easier to test the implementation this way, so having a separate type might be good.

Reply:
I like the separation tbh, I don't think it hurts and makes this somewhat easier to read.